workflow-ui/container-android/src/main/java/com/squareup/workflow1/ui/modal/ModalContainer.kt

```diff
@@ -167,12 +200,25 @@ public abstract class ModalContainer<ModalRenderingT : Any> @JvmOverloads constr
    * e.g. its content view. Not considered for equality.
    */
   @WorkflowUiExperimentalApi
-  protected data class DialogRef<ModalRenderingT : Any>(
-    val modalRendering: ModalRenderingT,
-    val viewEnvironment: ViewEnvironment,
-    val dialog: Dialog,
-    val extra: Any? = null
+  protected class DialogRef<ModalRenderingT : Any>(
+    public val modalRendering: ModalRenderingT,
+    public val viewEnvironment: ViewEnvironment,
+    public val dialog: Dialog,
+    public val extra: Any? = null
   ) {
+    /**
+     * The unique id of the `ViewTreeSavedStateRegistryOwner` that will be placed
+     * on the dialog's decor view by [stateRegistryAggregator].
+     */
+    internal lateinit var savedStateRegistryKey: String
+
+    public fun copy(
+      modalRendering: ModalRenderingT,
+      viewEnvironment: ViewEnvironment,
+    ): DialogRef<ModalRenderingT> = DialogRef(modalRendering, viewEnvironment, dialog, extra).also {
```

Comment:
Are we trying to enforce any behaviour here (i.e. specify `modalRendering` and `viewEnvironment` when making a new copy instance?). Otherwise we could just do a typical copy function like:

```
copy(
  modalRendering: ModalRenderingT = this.modalRendering,
  viewEnvironment: ViewEnvironment = this.viewEnvironment,
  dialog: Dialog = this.dialog,
  extra: Any? = this.extra,
```

Reply:
It won't matter in practice b/c this class is eliminated by the overhaul, but yeah, good point. I'll make it normal.

Reply:
Also note that that class itself is actually `protected`, so all of these `public`s are mistakes.

Reply:
Oh, actually required b/c it's no longer a `data` class.